Code/Framework/AzCore/Tests/Serialization/Json/AnySerializerTests.cpp

```diff
@@ -0,0 +1,281 @@
+                    {
+
+                    }
+                    else if (requestSpec.objectStatus == DefaultSerializedObjectStatus::Partial)
+                    {
+
+                    }
+                    else
+                    {
+                        return m_jsonForFullyConfiguredInstanceWithoutDefaults.begin();
+                    }
+                }
+            }
+
+            return AZStd::nullopt;
+        }
+
+        AZStd::string_view GetJsonForPartialDefaultInstance() override
+        {
+            return m_jsonForPartialDefaultInstanceStrippedDefaults.begin();
+        }
+
+        AZStd::string_view GetJsonForFullySetInstance() override
+        {
+            return m_jsonForFullyConfiguredInstance.begin();
+        }
+
+        
+        void ConfigureFeatures(JsonSerializerConformityTestDescriptorFeatures& features) override
+        {
+            features.EnableJsonType(rapidjson::kObjectType);
+            features.m_defaultIsEqualToEmpty = false;
+            features.m_partialDefaultReportingIsStrict = PartialDefaultReportingIsStrict;
+            features.m_supportsPartialInitialization = SupportsPartialDefaults;
+        }
+
+        bool AreEqual(const AZStd::any& lhs, const AZStd::any& rhs) override
+        {
+            if (lhs.type() == azrtti_typeid<T>() && lhs.type() == rhs.type())
+            {
+                auto lhsValue = AZStd::any_cast<const T>(&lhs);
+                auto rhsValue = AZStd::any_cast<const T>(&rhs);
+                return lhsValue && rhsValue && lhsValue->Equals(*rhsValue, m_fullyReflected);
+            }
+
+            return lhs.empty() == rhs.empty()
+                && lhs.type() == rhs.type();
+        }
+
+        void SetUp() override
+        {
+            auto instanceWithoutDefaults = T::GetInstanceWithoutDefaults();
+            auto instanceWithSomeDefaults = T::GetInstanceWithSomeDefaults();
+
+            const char* typeName = T::RTTI_TypeName();
+            const char* fullInstanceWithoutDefaultsInJSON = instanceWithoutDefaults.m_jsonWithStrippedDefaults;
+            const char* fullInstanceWithDefaultsInJSON = instanceWithoutDefaults.m_jsonWithKeptDefaults;
+            const char* withSomeDefaults = instanceWithSomeDefaults.m_jsonWithStrippedDefaults;
+            const char* withKeptDefaults = instanceWithSomeDefaults.m_jsonWithKeptDefaults;
+
+            azsnprintf(m_jsonForFullyConfiguredInstance.begin(), 2048, ANY_JSON_REFERENCE_STRING_FORMAT, typeName, fullInstanceWithDefaultsInJSON);
+            azsnprintf(m_jsonForFullyConfiguredInstanceWithoutDefaults.begin(), 2048, ANY_JSON_REFERENCE_STRING_FORMAT, typeName, fullInstanceWithoutDefaultsInJSON);
+            azsnprintf(m_jsonForPartialDefaultInstanceStrippedDefaults.begin(), 2048, ANY_JSON_REFERENCE_STRING_FORMAT, typeName, withSomeDefaults);
+            azsnprintf(m_jsonForPartialDefaultInstanceKeptDefaults.begin(), 2048, ANY_JSON_REFERENCE_STRING_FORMAT, typeName, withKeptDefaults);
```

Comment:
This should use` AZStd::fixed_string<2048>::format`.

The modifies the fixed_string, but it doesn't correctly set it's size because it is circumventing the class